pennylane/labs/trotter/realspace/matrix.py

```diff
@@ -0,0 +1,183 @@
+"""Functions for transforming a VibronicMatrix into a scipy csr matrix"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Tuple, Union
+
+import numpy as np
+import scipy as sp
+
+if TYPE_CHECKING:
+    from pennylane.labs.trotter.realspace import RealspaceOperator, RealspaceSum
+
+
+def position_operator(
+    gridpoints: int, sparse: bool = False, basis: str = "realspace"
+) -> Union[np.ndarray, sp.sparse.csr_array]:
+    """Returns a discretization of the position operator"""
+
+    if basis == "realspace":
+        matrix = _realspace_position(gridpoints)
+        return sp.sparse.csr_array(matrix) if sparse else matrix
+
+    if basis == "harmonic":
+        matrix = _harmonic_position(gridpoints)
+        return sp.sparse.csr_array(matrix) if sparse else matrix
+
+    raise ValueError(f'"{basis}" is not a valid basis')
+
+
+def _realspace_position(gridpoints: int) -> np.ndarray:
+    values = (np.arange(gridpoints) - gridpoints / 2) * (np.sqrt(2 * np.pi / gridpoints))
+    return np.diag(values)
+
+
+def _harmonic_position(gridpoints: int) -> np.ndarray:
+    rows = np.array(list(range(1, gridpoints)) + list(range(0, gridpoints - 1)))
+    cols = np.array(list(range(0, gridpoints - 1)) + list(range(1, gridpoints)))
+    vals = np.array([np.sqrt(i) for i in range(1, gridpoints)] * 2)
+
+    matrix = np.zeros(shape=(gridpoints, gridpoints))
+    matrix[rows, cols] = vals
+
+    return 1 / np.sqrt(2) * matrix
+
+
+def momentum_operator(
+    gridpoints: int, sparse: bool = False, basis: str = "realspace"
+) -> Union[np.ndarray, sp.sparse.csr_array]:
+    """Returns a discretization of the momentum operator"""
+
+    if basis == "realspace":
+        matrix = _realspace_momentum(gridpoints)
+        return sp.sparse.csr_array(matrix) if sparse else matrix
+
+    if basis == "harmonic":
+        matrix = _harmonic_momentum(gridpoints)
+        return sp.sparse.csr_array(matrix) if sparse else matrix
+
+    raise ValueError(f'"{basis}" is not a valid basis')
+
+
+def _realspace_momentum(gridpoints: int) -> np.ndarray:
+    values = np.arange(gridpoints)
+    values[gridpoints // 2 :] -= gridpoints
+    values = values * (np.sqrt(2 * np.pi / gridpoints))
+    dft = sp.linalg.dft(gridpoints, scale="sqrtn")
+    matrix = dft @ np.diag(values) @ dft.conj().T
+
+    return matrix
+
+
+def _harmonic_momentum(gridpoints: int) -> np.ndarray:
+    rows = np.array(list(range(1, gridpoints)) + list(range(0, gridpoints - 1)))
+    cols = np.array(list(range(0, gridpoints - 1)) + list(range(1, gridpoints)))
+    vals = np.array(
+        [np.sqrt(i) for i in range(1, gridpoints)] + [-np.sqrt(i) for i in range(1, gridpoints)]
+    )
+
+    matrix = np.zeros(shape=(gridpoints, gridpoints))
+    matrix[rows, cols] = vals
+
+    return (1j / np.sqrt(2)) * matrix
+
+
+def creation_operator(gridpoints: int, sparse: bool = False) -> Union[np.ndarray, sp.sparse.array]:
```

Comment:
We have other creation operators in PL, e.g., fermionic, bosonic. Please make sure that you don't use generic names. All user-facing functions will appear in the docs.